### PR TITLE
Basic utils and services for auth.

### PR DESCRIPTION
**Pull request by sumitdhungana14**
Basic token and auth services have been added.

### CODE REVIEW

**Review by jungRoit — APPROVED** [review]

File: src/services/auth/auth.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * @param {string} refreshToken
+ *
+ * @returns {Promise} A promise to a get request to get a new access token.
+ */
+export const refreshAccessToken = async (refreshToken) => {
+    try {
+        const response = await axios.post(`${config.baseURL}/oauth/access_token`, {
+            refreshToken
+        });
+        return mapLoginResponse(response);
```
Comment (su-sh) [inline src/services/auth/auth.js:59]: Let's add a new line before the return statement.

File: src/services/token.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { get, set, remove } from '../utils/storage';
+import * as constant from '../constant/constant';
```
Comment (su-sh) [inline src/services/token.js]: Let's order the imports.

This plugin might be handy:
https://marketplace.visualstudio.com/items?itemName=pratishshr.bharyang-vscode

File: src/services/token.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ * Gets access and refresh token from storage.
+ *
+ * @returns {Object}
+ */
+export const getAuthTokens = () => {
+    return {
+        accessToken: get(constant.ACCESS_TOKEN),
+        refreshToken: get(constant.REFRESH_TOKEN)
+    };
+};
```
Comment (su-sh) [inline src/services/token.js]: Shall we use a shorthand return?

```
export const getAuthTokens = () => ({
  accessToken: get(constant.ACCESS_TOKEN),
  refreshToken: get(constant.REFRESH_TOKEN),
});
```

File: src/utils/storage.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import localStorage from 'localStorage';
+
+/**
+ * Get value from local storage.
+ *
+ * @param {String} key : name of key to get value.
```
Comment (su-sh) [inline src/utils/storage.js:6]: Return doc missing here.

File: src/utils/storage.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+ */
+export function get(key) {
+    return localStorage.getItem(key);
```
Comment (su-sh) [inline src/utils/storage.js]: Let's use an arrow function.

I.e.
```
export const get = (key) => localStorage.getItem(key);
```

File: src/utils/storage.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * Set value to localStorage.
+ *
+ * @param {String} key : name of key to set value.
+ * @param {*} value : value to set.
+ */
+export function set(key, value) {
+    localStorage.setItem(key, value);
+}
```
Comment (su-sh) [inline src/utils/storage.js]: .

File: src/utils/storage.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * Removes value from localStorage.
+ *
+ * @param {string} key The key to remove.
+ */
+export function remove(key) {
+    localStorage.removeItem(key);
+}
```
Comment (su-sh) [inline src/utils/storage.js]: .

File: src/utils/interceptor.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
+import axios from 'axios';
+
+import { getAuthTokens,setAccessToken } from '../services/token';
+import {refreshAccessToken} from '../services/auth/auth';
+import config from '../config';
```
Comment (su-sh) [inline src/utils/interceptor.js]: Let's order the imports

File: src/utils/http.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Axios get method.
+ *
+ * @param {string} url
+ * @returns {Promise}
+ */
+export function get(url) {
+    return instance.get(url);
+}
```
Comment (su-sh) [inline src/utils/http.js]: Let's use an arrow function.

File: src/services/auth/auth.js [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+import { clearAuthTokens } from '../token';
+
+import config from '../../config';
+
+/**
+ * Maps to token object.
+ *
+ * @param {Object} response
+ *
```
Comment (su-sh) [inline src/services/auth/auth.js]: I don't think an empty line is necessary before a return tag in JSDoc.
Let's remove this empty line, and make the JSDoc consent within the repo.

**Comment by su-sh** [comment]
@sumitdhungana14 the changes looks good, few comments added can you please look at it?

**Review by su-sh — COMMENTED** [review]
.

**Review by su-sh — APPROVED** [review]
LGTM 👍 

Thanks for the contribution @ishan-me

**Pull request merged**